depends: qt: Fix {C{,XX},LD}FLAGS pickup

dongcarl opened the pull request:
Note: ~~Will~~ [Did](https://github.com/bitcoin/bitcoin/issues/16838) open issue about robustness of `sed` calls in `depends`.

dongcarl committed:
depends: qt: Fix C{,XX}FLAGS pickup

laanwj commented on depends/packages/qt.mk:
```diff
@@ -178,9 +178,9 @@ define $(package)_preprocess_cmds
   patch -p1 -i $($(package)_patch_dir)/no-xlib.patch &&\
   echo "QMAKE_LINK_OBJECT_MAX = 10" >> qtbase/mkspecs/win32-g++/qmake.conf &&\
   echo "QMAKE_LINK_OBJECT_SCRIPT = object_script" >> qtbase/mkspecs/win32-g++/qmake.conf &&\
-  sed -i.old "s|QMAKE_CFLAGS            = |!host_build: QMAKE_CFLAGS            = $($(package)_cflags) $($(package)_cppflags) |" qtbase/mkspecs/win32-g++/qmake.conf && \
```
FWIW if you use bash as shell, you need to escape the `!` here to not get a `-bash: !host_build: event not found` error
(I tried executing this command separately to test it)

dongcarl replied:
Should we explicitly set `SHELL` then? The default is `/bin/sh` I think.

dongcarl replied:
Ah gotcha, fun fact: when `argv[0]` looks like it's `sh`, I believe `bash` will be in posix-compliant mode.

laanwj commented on depends/packages/qt.mk:
```diff
@@ -178,9 +178,9 @@ define $(package)_preprocess_cmds
   echo "QMAKE_LINK_OBJECT_SCRIPT = object_script" >> qtbase/mkspecs/win32-g++/qmake.conf &&\
-  sed -i.old "s|QMAKE_CFLAGS            = |!host_build: QMAKE_CFLAGS            = $($(package)_cflags) $($(package)_cppflags) |" qtbase/mkspecs/win32-g++/qmake.conf && \
+  sed -i.old "s|QMAKE_CFLAGS           += |!host_build: QMAKE_CFLAGS            = $($(package)_cflags) $($(package)_cppflags) |" qtbase/mkspecs/win32-g++/qmake.conf && \
   sed -i.old "s|QMAKE_LFLAGS            = |!host_build: QMAKE_LFLAGS            = $($(package)_ldflags) |" qtbase/mkspecs/win32-g++/qmake.conf && \
```
doesn't look like this line is working, there's no bare "QMAKE_LFLAGS" in my `win32-g++/qmake.conf` at all (qt 5.9.7)

dongcarl replied:
Digging a little into this... It would seem that `QMAKE_LFAGS` was removed in https://github.com/qt/qtbase/commit/39fc377bf105ba09e2a8f9acae467dc789b96525, and first made it into qt `v5.9.0`. We can just add `QMAKE_LFLAGS            = $($(package)_ldflags)` back in though.

dongcarl replied:
See aaec508742b89ec7b2c0a6d5a37b871c726d2598 for my fix.

laanwj replied:
Looks good to me, thanks for addressing this

dongcarl committed:
depends: qt: Fix LDFLAGS pickup

QMAKE_LFLAGS was removed from qtbase/mkspecs/win32-g++/qmake.conf in
39fc377bf105ba09e2a8f9acae467dc789b96525. Here, we add it back in with
our LDFLAGS from depends before the first occurance of any
QMAKE_LFLAGS_* variable settings.

laanwj commented:
Requested a new gitian build. ACK from me if it passes.

laanwj commented:
ACK 1b4030e26450b0f9320d64714d8e8dee37b050d2
